.github/workflows/run-tests.yml

```diff
@@ -0,0 +1,81 @@
+name: Path tests
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  linter:
+    name: Run golangci-lint
+    runs-on: "ubuntu-22.04"
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - uses: actions/setup-go@v5
+        with:
+          go-version: "1.22.3"
+
+      - name: Setup Git Authentication
+        run: |
+          git config --global url."https://${{ github.token }}:x-oauth-basic@github.com/".insteadOf "https://github.com/"
+
+      - name: Run golangci-lint
+        uses: golangci/golangci-lint-action@v6
+        with:
+          version: latest
```

Comment:
I wonder if we should peg this to a set version as well?

Reply:
I think it's ok if we keep using `latest` since it's the golangci version and we're not pairing this with specific golang installation.